legal-api/src/legal_api/models/business.py

```diff
@@ -179,15 +179,18 @@ def get_ar_dates(self, next_ar_year):
             # is required. We need more discussion to understand different scenario's which can come across in future.
             if next_ar_year == 2020:
                 # For year 2020, set the max date as October 31th next year (COVID extension).
-                ar_max_date = datetime(next_ar_year, 10, 31).date()
+                ar_max_date = datetime(next_ar_year + 1, 10, 31).date()
             else:
                 # If this is a CO-OP, set the max date as April 30th next year.
-                ar_max_date = datetime(next_ar_year, 4, 30).date()
+                ar_max_date = datetime(next_ar_year + 1, 4, 30).date()
         elif self.legal_type == self.LegalTypes.BCOMP.value:
             # For BCOMP min date is next anniversary date.
-            ar_min_date = datetime(next_ar_year, self.next_anniversary.month, self.next_anniversary.day).date()
+            ar_min_date = datetime(next_ar_year, self.founding_date.month, self.founding_date.day).date()
```

Comment:
next_anniversary is calculated as: 
```python
    def next_anniversary(self):
        """Retrieve the next anniversary date for which an AR filing is due."""
        last_anniversary = self.founding_date
        if self.last_ar_date:
            last_anniversary = self.last_ar_date

        return last_anniversary + datedelta.datedelta(years=1)
```
what's wrong with that?
```suggestion
            ar_min_date = datetime(next_ar_year, self.next_anniversary.month, self.next_anniversary.day).date()
```

Reply:
I thought that nextAnnualReport should be the date for the next_year to be filed and should change in every `task` element in the json as do arMinDate and arMaxDate. Otherwise it will be repeated in all `task` elements and should equal to the least arMinDate.

I'm gonna revert it to next_anniversary then.